mods/taskbar-empty-space-clicks.wh.cpp

```diff
@@ -0,0 +1,1572 @@
+    if (!lpImmersiveShellHookService)
+        return NULL;
+
+    LONG_PTR lpImmersiveWindowMessageService =
+        *(LONG_PTR *)(lpImmersiveShellHookService + DO10_3264(0, 0, , , , , , , 0x88, 0xE0, , , , , , , 0xA0, 0x108, 0x98, 0x100));
+    if (!lpImmersiveWindowMessageService)
+        return NULL;
+
+    return *(HWND *)(lpImmersiveWindowMessageService + DO10_3264(0, 0, , , , , , , 0x4C, 0x80, , , 0x50, 0x88, , , 0x58, 0x98, 0x54, 0x90));
+}
+
+TaskBarAction ParseMouseActionSetting(const wchar_t *option) {
+    const auto value = Wh_GetStringSetting(option);
+    const auto equals = [](const wchar_t *str1, const wchar_t *str2) { return wcscmp(str1, str2) == 0; };
+
+    TaskBarAction action = ACTION_NOTHING;
+    if (equals(value, L"ACTION_NOTHING")) {
+        action = ACTION_NOTHING;
+    } else if (equals(value, L"ACTION_SHOW_DESKTOP")) {
+        action = ACTION_SHOW_DESKTOP;
+    } else if (equals(value, L"ACTION_ALT_TAB")) {
+        action = ACTION_ALT_TAB;
+    } else if (equals(value, L"ACTION_TASK_MANAGER")) {
+        action = ACTION_TASK_MANAGER;
+    } else if (equals(value, L"ACTION_MUTE")) {
+        action = ACTION_MUTE;
+    } else if (equals(value, L"ACTION_TASKBAR_AUTOHIDE")) {
+        action = ACTION_TASKBAR_AUTOHIDE;
+    } else if (equals(value, L"ACTION_WIN_TAB")) {
+        action = ACTION_WIN_TAB;
+    } else if (equals(value, L"ACTION_HIDE_ICONS")) {
+        action = ACTION_HIDE_ICONS;
+    } else {
+        Wh_Log(L"Error: unknown action '%s' for option '%s'!", value, option);
+        action = ACTION_NOTHING;
+    }
+    Wh_Log(L"Selected '%s' option %d", option, action);
+
+    return action;
+}
+
+void LoadSettings() {
+    g_settings.oldTaskbarOnWin11 = Wh_GetIntSetting(L"oldTaskbarOnWin11");
+    g_settings.doubleClickTaskbarAction = ParseMouseActionSetting(L"doubleClickAction");
+    g_settings.middleClickTaskbarAction = ParseMouseActionSetting(L"middleClickAction");
+}
+
+bool GetTaskbarAutohideState() {
+    if (g_hTaskbarWnd == NULL) {
+        return false;
+    }
+    APPBARDATA msgData{};
+    msgData.cbSize = sizeof(msgData);
+    msgData.hWnd = g_hTaskbarWnd;
+    LPARAM state = SHAppBarMessage(ABM_GETSTATE, &msgData);
+    return state & ABS_AUTOHIDE;
+}
+
+void SetTaskbarAutohide(bool enabled) {
+    if (g_hTaskbarWnd == NULL) {
+        return;
+    }
+
+    APPBARDATA msgData{};
+    msgData.cbSize = sizeof(msgData);
+    msgData.hWnd = g_hTaskbarWnd;
+    msgData.lParam = enabled ? ABS_AUTOHIDE : ABS_ALWAYSONTOP;
+    SHAppBarMessage(ABM_SETSTATE, &msgData);
+}
+
+bool FindDesktopWindow() {
+    HWND hParentWnd = FindWindow(L"Progman", NULL);     // Program Manager window
+	if (!hParentWnd) {
+        Wh_Log(L"Failed to find Progman window");
+        return false;
+    }
+
+	HWND hChildWnd = FindWindowEx(hParentWnd, NULL, L"SHELLDLL_DefView", NULL);     // parent window of the desktop
```

Comment:
Some tabs are lurking here.

Reply:
Still there, but not that important, you can fix it in the next version if you want.